src/libraries/System.Reflection.Emit.Lightweight/ref/System.Reflection.Emit.Lightweight.cs

```diff
@@ -30,13 +30,21 @@ public sealed partial class DynamicILInfo
     }
     public sealed partial class DynamicMethod : System.Reflection.MethodInfo
     {
+        [System.Diagnostics.CodeAnalysis.RequiresDynamicCode("Generating new code at runtime is not supported with native AOT.")]
```

Comment:
Do we want to use `"native AOT"` in the public message shown to users? That seems a bit too specific about the technology here. That's why we didn't name the attribute "NativeAOTUnsupported".

```suggestion
        [System.Diagnostics.CodeAnalysis.RequiresDynamicCode("Creating a DynamicMethod requires dynamic code.")]
```

Reply:
Note that this comment applies throughout the entire change. Not just this one place.